### PR TITLE
fix: don't let create end_date before start_date >= today

### PR DESCRIPTION
**Pull request by gabrielricardo7**

### CODE REVIEW

**Review by marcelosj3 — APPROVED** [review]

File: campaigns/utils.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from datetime import date
+
+from sos_brazil.exceptions import BadEndDateException, BadStartDateException
+
+
+def check_dates(start_date, end_date):
+    if start_date < date.today():
+        raise BadStartDateException()
+
+    elif end_date <= start_date:
```
Comment (marcelosj3) [inline campaigns/utils.py:10]: I believe, by our discussions, that the user can create an `end_date` equals to `start_date`, like an "one day campaign" thing

**Review by marcelosj3 — CHANGES_REQUESTED** [review]

**Review by gabrielsuch — APPROVED** [review]

File: campaigns/utils.py (hunk 1 above)
Comment (gabrielricardo7) [reply campaigns/utils.py]: fix: user can create and change end_date equals start_date

File: sos_brazil/exceptions.py [hunk 2]
```diff
@@ -59,3 +59,18 @@ def __init__(
...
+
+
+class BadEndDateException(APIException):
+    status_code = status.HTTP_400_BAD_REQUEST
+    default_detail = "The end_date must be later than the start_date."
+
+
+class CampaignDateException(APIException):
+    status_code = status.HTTP_400_BAD_REQUEST
+    default_detail = "The start_date can't be later or equal to the end_date."
```
Comment (marcelosj3) [inline sos_brazil/exceptions.py:76]: It can be equal to the end_date

File: campaigns/serializers.py [hunk 3]
```diff
@@ -29,13 +30,25 @@ def validate_goal(self, value):
...
 
         return campaign
 
     def update(self, instance, validated_data):
         non_updatable_keys = ["collected", "goal_reached"]
         wrong_keys = []
+        start_date = validated_data.get("start_date", instance.start_date)
+        end_date = validated_data.get("end_date", instance.end_date)
+
+        if start_date > end_date:
```
Comment (marcelosj3) [inline campaigns/serializers.py:48]: Couldn't you make this check inside the `check_dates` function?

File: campaigns/utils.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+
+def check_dates(start_date, end_date):
+    if start_date < date.today():
+        raise BadStartDateException()
+
+    elif end_date < start_date:
+        raise BadEndDateException()
+
+    return True
```
Comment (marcelosj3) [inline campaigns/utils.py:13]: Are we using this return for anything?

**Review by marcelosj3 — CHANGES_REQUESTED** [review]

**Review by marcelosj3 — APPROVED** [review]

**Pull request merged**